app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt

```diff
@@ -29,39 +30,42 @@ class ProfilePresenter @Inject constructor (private val view: ProfileView,
                 myselfId = myself.id
                 val avatarUrl = UrlHelper.getAvatarUrl(serverUrl, myself.username!!)
                 view.showProfile(
-                    avatarUrl,
-                    myself.name!!,
-                    myself.username!!,
-                    myself.emails?.get(0)?.address!!
+                        avatarUrl,
+                        myself.name!!,
+                        myself.username!!,
+                        myself.emails?.get(0)?.address!!
                 )
             } catch (exception: RocketChatException) {
                 exception.message?.let {
                     view.showMessage(it)
                 }.ifNull {
-                        view.showGenericErrorMessage()
-                    }
+                            view.showGenericErrorMessage()
+                        }
             } finally {
                 view.hideLoading()
             }
         }
     }
 
-    fun updateUserProfile(email: String, name: String, username: String, avatarUrl: String = "") {
+    fun updateUserProfile(email: String, name: String, username: String, avatarUrl: String = "", avatarImage: File?) {
         launchUI(strategy) {
             view.showLoading()
             try {
-                if(avatarUrl!="") {
+                if (avatarUrl != "") {
                     client.setAvatar(avatarUrl)
                 }
+                if (avatarImage != null)
+                    client.setAvatar(avatarImage, "image/jpeg")
```

Comment:
First you are [allowing to the user the selection of any image type](https://github.com/aniketsingh03/Rocket.Chat.Android/blob/b811400eec76aa399ccba84039f596fe02b6d9d1/app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt#L153) but here you are specifying that any image picked by the user is a JPEG image. 
We added an [extension function](https://github.com/aniketsingh03/Rocket.Chat.Android/blob/b811400eec76aa399ccba84039f596fe02b6d9d1/app/src/main/java/chat/rocket/android/util/extensions/Uri.kt#L44) to get the image type from an URI: 

 `yourUri.getMimeType(aContext)`

Could you change it?